Add contrail configuration for salt

ananth-at-camphor-networks opened the pull request.

justinsb committed:
AWS: Fix race-condition in tagging of security group

When we create a security-group in the AWS API, there is sometimes
a delay before we can tag it (the AWS API is eventually consistent).

So we wrap CreateTags in a simple retry loop.

justinsb committed:
AWS: Fix race in security-group read/create

We need to find the ID for a named security group, or create a new one.

We do this by listing the security groups, and then doing a create if we
cannot find one.  This is a race though; against another thread if the
AWS API were consistent, but generally because the AWS API is actually
eventually consistent.

We wrap it in a retry loop.

jfchevrette committed:
restart network twice to workaround bug

jfchevrette committed:
properly make sure that eth1 is not managed by NetworkManager

jfchevrette committed:
same change for minion provision script

Commit:
Refactor the functions that generate auth for the kubelet and kubeproxy and remove the insecure configuration now that GKE has plumbed through certificates.

sgotti committed:
Set the right master port to kube-proxy in libvirt-coreos.

The currently specified port is the old 7080 read-only-port. As the
libvirt-coreos doesn't configure security, change it to the insecure-port 8080.

mikedanese committed:
Revert "Revert "Fix libvirt-coreos cluster""

This reverts commit 889d724c1996e400ce01fcf397644a9df72f36a0.

mikedanese committed:
fix test with array expansion

justinsb committed:
Fixes per review

Primarily go style issues; also a TODO that really exponential backoff
is the correct policy for API call retries.

iterion committed:
Fetch AWS PrivateDNSName for route recon

Commit:
Register the kubelet on the master node with an apiserver. This option is separated from the apiserver running locally on the master node so that it can be optionally enabled or disabled as needed.

Also, fix the healthchecking configuration for the master components, which
was previously only working by coincidence:

If a kubelet doesn't register with a master, it never bothers to figure out
what its local address is. In which case it ends up constructing a URL like
http://:8080/healthz for the http probe. This happens to work on the master
because all of the pods are using host networking and explicitly binding to
127.0.0.1. Once the kubelet is registered with the master and it determines
the local node address, it tries to healthcheck on an address where the pod
isn't listening and the kubelet periodically restarts each master component
when the liveness probe fails.

a-robinson committed:
Collect more of the VM's log files (monit, salt) in fluentd-gcp.

a-robinson committed:
Collect more of the VM's log files (monit, salt) in fluentd-es.

pedro-r-marques commented on cluster/addons/dns/skydns-rc.yaml.in:
```diff
@@ -48,6 +48,7 @@ spec:
         args:
         # command = "/kube2sky"
         - -domain={{ pillar['dns_domain'] }}
+        - -kube_master_url=http://10.0.0.1:8080
```
I'm assuming that 10.0/16 is the portal_net for you ?
We would have to get this parameter from the "pillar".

pedro-r-marques commented on cluster/addons/dns/skydns-svc.yaml.in:
```diff
@@ -11,6 +11,7 @@ spec:
   selector:
     k8s-app: kube-dns
   clusterIP:  {{ pillar['dns_server'] }}
+  deprecatedPublicIPs: ["10.1.4.53"]
```
Aren't the yaml.in files processed by salt. We should be able to do the transformation via salt (so that it is conditional).
There must be an salt option for wether or not we are using contrail.

pedro-r-marques commented on cluster/addons/kube-ui/kube-ui-endpoint.yaml:
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
```
The issue at the moment is that the /api/proxy URLs are using the endpoints rather that the service address.
We may be able to upstream a change to  use the service address as the URL of the proxy. So this is a temporary hack.

pedro-r-marques commented on cluster/saltbase/salt/contrail-apiserver/init.sls:
```diff
@@ -0,0 +1,8 @@
+/etc/kubernetes/manifests/contrail-apiserver.manifest:
+  file.managed:
+    - source: https://raw.githubusercontent.com/pedro-r-marques/contrail-kubernetes/manifests/cluster/contrail-api.manifest
```
These files have merged into the Juniper repository.

pedro-r-marques commented on cluster/saltbase/salt/kubelet/default:
```diff
@@ -55,7 +55,7 @@
 {% set cluster_domain = "" %}
 {% if pillar.get('enable_cluster_dns', '').lower() == 'true' %}
-  {% set cluster_dns = "--cluster_dns=" + pillar['dns_server'] %}
+  {% set cluster_dns = "--cluster_dns=" + "10.1.4.53" %}
```
pillar['dns_server'] is correct and should be used above.

pedro-r-marques commented on cluster/saltbase/salt/top.sls:
```diff
@@ -33,6 +34,14 @@ base:
     - kube-controller-manager
     - kube-scheduler
     - monit
+    - contrail-apiserver
```
These will have to be conditional.d

ananth-at-cloud-spectra-dot-ai replied:
Yes. Many of the hard-coded stuff need to be cleaned up. I wanted to give
Sanju a heads up.
10.0.0.0/16 is portal network in my aws setup.

Regards,
Ananth

On Tue, Aug 4, 2015 at 6:32 PM, Pedro Marques notifications@github.com
wrote:

> In cluster/saltbase/salt/top.sls
> https://github.com/rombie/kubernetes/pull/1#discussion_r36262241:
> 
> > @@ -33,6 +34,14 @@ base:
> >      - kube-controller-manager
> >      - kube-scheduler
> >      - monit
> > -    - contrail-apiserver
> 
> These will have to be conditional.d
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/rombie/kubernetes/pull/1/files#r36262241.

vishh committed:
Updating heapster to v0.16.1.

alex-mohr committed:
Merge pull request #12237 from jbeda/automated-cherry-pick-of-#11390-upstream-release-1.0

Automated cherry pick of #11390 upstream release 1.0

alex-mohr committed:
Merge pull request #12244 from a-robinson/automated-cherry-pick-of-#12240-upstream-release-1.0

Auto commit by PR queue bot

erictune committed:
troubleshooting for misplaced/misspelled fields

mikedanese committed:
rewrite go imports for godeps

mikedanese committed:
fix build

mikedanese committed:
fixup development doc for new vanity path

a-robinson committed:
Merge pull request #12243 from vishh/v0.1.0

Updating heapster to v0.16.1.

a-robinson committed:
Bump the heapster pod's memory limit from 200MiB to 300MiB.

a-robinson committed:
Merge pull request #12318 from a-robinson/automated-cherry-pick-of-#12314-upstream-release-1.0

Bump the heapster pod's memory limit from 200MiB to 300MiB.

brendandburns committed:
Add support for supervisord as a monit alternative.

devurandom committed:
Fix link to etcd on quay.io in from-scratch-guide

wojtek-t committed:
Merge pull request #12336 from devurandom/patch-1

Fix link to etcd on quay.io in from-scratch-guide

zmerlynn committed:
Update GCE to container-vm-v20150806

Also changes the node image to default to master image

zmerlynn committed:
Merge pull request #12367 from roberthbailey/automated-cherry-pick-of-#11483-#12349-upstream-release-1.0

Automated cherry pick of #11483 #12349

brendandburns committed:
Merge pull request #12369 from brendandburns/automated-cherry-pick-of-#12325-release-1.0

Automated cherry pick of #12325

dchen1107 committed:
Merge pull request #12370 from zmerlynn/automated-cherry-pick-of-#12366-upstream-release-1.0

Automated cherry pick of #12366

brendandburns committed:
Merge pull request #12358 from erictune/automated-cherry-pick-of-#12296-upstream-release-1.0

Automated cherry pick of #12296

brendandburns committed:
Versioning docs and examples for 1.0.3

brendandburns committed:
Kubernetes version v1.0.3

zmerlynn committed:
Merge pull request #12382 from brendandburns/release-1.0

Release 1.0.3

derekparker committed:
Handle full hostname when computing host tag on GCE

The current code assumes the full domain name will not be included,
which is not always the case. This patch adds support for computing the
host tag from a fully qualified domain name.

derekwaynecarr committed:
Fix vagrant kube-up

dchen1107 committed:
Fix kube-push using gcloud compute instance-groups managed set-instance-template`

brendandburns committed:
Fix the node controller in an emulation of #12546

mikedanese committed:
point shippable to new repository

mikedanese committed:
rewrite go imports

mikedanese committed:
run gofmt on everything we touched

zmerlynn committed:
Merge pull request #12817 from kubernetes/automated-cherry-pick-of-#12310-#12592-upstream-release-1.0

Automated cherry pick of #12310 #12592 upstream release 1.0

Commit:
Don't register the master node into the cluster by default.

wojtek-t committed:
Stop using deprecated commands

zmerlynn committed:
Merge pull request #12928 from mikedanese/automated-cherry-pick-of-#11794-upstream-release-1.0

Automated cherry pick of #11794 upstream release 1.0

zmerlynn committed:
Merge pull request #12924 from rubenv/automated-cherry-pick-of-#11804-upstream-release-1.0

Automated cherry pick of #11804

zmerlynn committed:
Merge pull request #12930 from mikedanese/automated-cherry-pick-of-#11029-upstream-release-1.0

Automated cherry pick of #11029 upstream release 1.0

sgallagher committed:
Ensure that kube-apiserver starts after etcd

If etcd is running on the same system as kube-apiserver and is
being started as part of the same series of services (such as with a
custom systemd target), we need to ensure that it orders later in
startup than etcd or it will fail to find the local service.

Note: this does not attempt to start etcd on the local system if it
is not already part of the transaction, so it will have no effect
if someone calls `systemctl start kube-apiserver.service` directly.

zmerlynn committed:
Merge pull request #12559 from brendandburns/cherrypick

Pseudo Cherrypick: Fix the node controller in an emulation of #12546

zmerlynn committed:
Merge pull request #12591 from dchen1107/automated-cherry-pick-of-#12541-upstream-release-1.0

Automated cherry pick of #12541

j3ffml committed:
Future-proof detect-minions for gke

saad-ali committed:
Merge pull request #12939 from sgallagher/release-1.0

Ensure that kube-apiserver starts after etcd

roberthbailey committed:
Merge pull request #12997 from zmerlynn/automated-cherry-pick-of-#12840-upstream-release-1.0

Automated cherry pick of #12840

zmerlynn committed:
Merge pull request #13013 from roberthbailey/automated-cherry-pick-of-#12897-#13001-upstream-release-1.0

Automated cherry pick of #12897 #13001 upstream release 1.0

j3ffml committed:
fix mig command for gce

zmerlynn committed:
Merge pull request #13059 from roberthbailey/automated-cherry-pick-of-#13046-upstream-release-1.0

Automated cherry pick of #13046

zmerlynn committed:
Merge pull request #13043 from justinsb/automated-cherry-pick-of-#12029-upstream-release-1.0

Automated cherry pick of #12029

bfallik committed:
Update kubelet.md

fix simple typo

mwielgus committed:
Merge pull request #13067 from bfallik/patch-2

Update kubelet.md (fix small typo)

a-robinson committed:
Be more lenient when deriving the node tag from a node name on GCE.

roberthbailey committed:
Merge pull request #13127 from a-robinson/automated-cherry-pick-of-#12438-#13121-upstream-release-1.0

Automated cherry pick of #12438 #13121 upstream release 1.0

zmerlynn committed:
Versioning docs and examples for 1.0.4

zmerlynn committed:
Kubernetes version v1.0.4

roberthbailey committed:
Merge pull request #13159 from zmerlynn/release-1.0

Release 1.0.4

zmerlynn committed:
Revert "Ensure that kube-apiserver starts after etcd"

zmerlynn committed:
Merge pull request #13170 from kubernetes/revert-12939-release-1.0

Revert "Ensure that kube-apiserver starts after etcd"

zmerlynn committed:
Revert "Revert "Ensure that kube-apiserver starts after etcd""

zmerlynn committed:
Merge pull request #13206 from kubernetes/revert-13170-revert-12939-release-1.0

Revert "Revert "Ensure that kube-apiserver starts after etcd""

roberthbailey committed:
Merge pull request #13224 from derekwaynecarr/automated-cherry-pick-of-#12514-upstream-release-1.0

Automated cherry pick of #12514

ananth-at-camphor-networks committed:
Add contrail setup scripts
